### ISSUE TITLE
Field PATH env meaning

### ISSUE THREAD

**Issue by Chris2011** [joakim-eriksson/nb-eslint#12]
Can you please explain, what that input field mean and for what it is?

![image](https://user-images.githubusercontent.com/795658/46017009-c4ace680-c0d6-11e8-8d9b-3f6cf556b13a.png)

**Comment by Chris2011** [comment]
The other thing, I  use windows, this is confusing, why some linux stuff is inside. I know it is a sample, for there should be no sample or the sample inside the comment below or for each operating system a sample.

**Comment by pedrohidalgo** [comment]
The Path environment variable will be used when running the eslint command. 

On macos there was a problem that the PATH environment variable wasn't set and therefore the node executable was not detected.

I understand this is not needed for eveybody but It was a workaround for me as a Macos User.

**Comment by Chris2011** [comment]
@pedrohidalgo but then you should still have problems with other CLIs like node, npm, bower, etc. Why is it explicit eslint?

**Comment by Chris2011** [comment]
You should set your path variable in your OS, not inside of the IDE.

**Comment by pedrohidalgo** [comment]
> @pedrohidalgo but then you should still have problems with other CLIs like node, npm, bower, etc. Why is it explicit eslint?

Because ESLint depends on Node. When ESLint was running the problem was that the node executable was not found.

> You should set your path variable in your OS, not inside of the IDE.

The path variable is set on my OS, the problem is that the plugin doesn't have access to the PATH environment variable from the OS at least on Macos. That's why I had to set it myself using `builder.getEnvironment().setVariable("PATH", pathEnvVar);`

**Comment by Chris2011** [comment]
Any other CLI like Angular CLI or npm or bower depends on node too. So this is why I don't get it, because bower, npm should make the same problems at you. You can try to use this plugin too and tell me, whether you have the same problem or not, if you want: https://github.com/Chris2011/NBAngularCLI/releases/tag/1.0.1

**Comment by pedrohidalgo** [comment]
It seems to me that this problem only happens on macos. I have 2 other plugins ( [nbshell](https://github.com/pedrohidalgo/nbshell) and [netbeans_shell_commands_executor](https://github.com/pedrohidalgo/netbeans_shell_commands_executor) ) that use the `ExternalProcessBuilder ` class to execute commands in the OS and I had to specify the PATH variable on both in order for them to work correctly.

btw, unfortunately, I don't use angular CLI so can't test your plugin now.

**Comment by Chris2011** [comment]
then test npm or bower inside of node or sass/less, because they are official.

**Comment by pedrohidalgo** [comment]
> then test npm or bower inside of node or sass/less, because they are official.

If you're talking about testing inside a node project? it works. But when executing the commands on netbeans plugin it fails on macos because the PATH variable is not detected by `ExternalProcessBuilder `

**Comment by Chris2011** [comment]
So running NPM inside of NetBeans also has the problem on MacOS, this is what you are saying, right? If yes, fine, I only wanted to make sure, that this is a general problem.

**Comment by Chris2011** [comment]
Maybe you can have a look into the old bugzilla and finding similar, I can't understand, that this problem is that unknown.

**Comment by pedrohidalgo** [comment]
> So running NPM inside of NetBeans also has the problem on MacOS, this is what you are saying, right? If yes, fine, I only wanted to make sure, that this is a general problem.

Inside netbeans in the terminal, there's no problem, it works correctly. **But** when a **NetBeans Plugin** try to execute the commands using [http://bits.netbeans.org/8.2/javadoc/org-netbeans-modules-extexecution/org/netbeans/api/extexecution/ExternalProcessBuilder.html](http://bits.netbeans.org/8.2/javadoc/org-netbeans-modules-extexecution/org/netbeans/api/extexecution/ExternalProcessBuilder.html) then the command can fail because the environment PATH variable is not detected.

**Comment by pedrohidalgo** [comment]
Did you understand that this problem is happening only when executing commands internally in the plugin?

**Comment by Chris2011** [comment]
Yes, but I mean not via Terminal, start using the NetBeans actions like `npm install` or `npm scripts`, because they are using the native NetBeans actions/methods and if this is working correctly under MacOS then there is a problem with the plugin as you mentioned.

And AFAIK the ExternalProcessBuilder is deprecated. I know back in the days, when I did smth similar there was such deprecation message for ExternalProcessBuilder. So that means, the plugin needs to take another approach, which is already done. ESLint uses ProcessBuilder.getLocal(); https://github.com/joakim-eriksson/nb-eslint/blob/master/src/se/jocke/nb/eslint/ESLint.java#L101

So is this still a problem with using ProcessBuilder.getLocal() instead of an ExternalProcessBuilder?

**Comment by pedrohidalgo** [comment]
NetBeans actions like `npm scripts` from the contextual menu of a project node in the projects window works correctly.

> So is this still a problem with using ProcessBuilder.getLocal() instead of an ExternalProcessBuilder?

I don't know exactly, I didn't migrate my code to use that yet. But probably it is still a problem since nb-eslint is using `ProcessBuilder.getLocal()` and the problem was happening on macos. That's why I created a PR on this plugin

**Comment by Chris2011** [comment]
You can test the latest version of this plugin on mac, because it is still implemented and say us, whether it is working or not.

**Comment by pedrohidalgo** [comment]
Just Did. And it is working correctly. Because I specified the PATH environment variable using the textfield provided.

**Comment by Chris2011** [comment]
No, I mean w/o using that field. I want to verify the original problem, not the solution with PATH thingy in NetBeans.

Is it working, with an empty field of the path variable? I don't know how much the path variable code is binded into the plugin and works w/o any value.

**Comment by pedrohidalgo** [comment]
No, it is not working. 

The original error I got when I debugged this problem on may last year was something indicating that the node executable couldn't be found.

**Comment by pedrohidalgo** [comment]
This is the exact line when the variable is set: [https://github.com/joakim-eriksson/nb-eslint/blob/master/src/se/jocke/nb/eslint/ESLint.java#L106](https://github.com/joakim-eriksson/nb-eslint/blob/master/src/se/jocke/nb/eslint/ESLint.java#L106)

**Comment by Chris2011** [comment]
> No, it is not working.
> 
> The original error I got when I debugged this problem on may last year was something indicating that the node executable couldn't be found.

Which error do you get now? Exactly now, when you test it w/o using the PATH var. As I said the code seems rewritten. I think it is not needed at all, because the node/npm/bower stuff is still working from within the IDE w/o any problems.

**Comment by Chris2011** [comment]
The other thing, what came into my mind right now is, that you have to add the full path into the input field, where you click on browse. So it doesn't matter that you have it inside PATH, you add the whole path with the executable to that field. See my screenshots, for what I mean.

![image](https://user-images.githubusercontent.com/795658/51390184-deaf8080-1b2e-11e9-8de8-0885a148e1eb.png)

I know that this is windows, but if you click on search (It wasn't there before) you should get the full path of it. Maybe smth like this: `/usr/local/opt/nvm/versions/node/v0.12.2/lib/node_modules/eslint` I don't know how it will look like in Mac but maybe smth like that. So for bower you have the full path on windows, as you can see it here. Maybe you can share a screenshot, to show us, what is inside of the first input field to set the executable. Because I got the full path of that file. This is why I don't need to set the PATH variable (Still to say that I know that I'm under windows, but it is the full path, not only `eslint`.

**Comment by Chris2011** [comment]
![image](https://user-images.githubusercontent.com/795658/51391692-099bd380-1b33-11e9-9403-6c8fa90dcef8.png)
This is my option.

**Comment by Chris2011** [comment]
@pedrohidalgo anything new here, from your side? Couldn't test it yet on a mac, will test it later. What did you add into the ESLint: field? Only the command "eslint" or the full path? Because this is what it was added for. Adding the full path to the executable, not only the command.

**Comment by Chris2011** [comment]
Will try to find a mac to test it.

**Comment by Chris2011** [comment]
Luckily, today I had a chance to test my plugin NBAngularCLI and NB-eslint on a mac OS Sierra (10.12.6). For my cli, that I installed globally via `npm i -g @angular/cli` NetBeans found this path when I click on search: `/usr/local/bin/ng` and after choosing an angular CLI project, it works. It created the project. For NB-ESlint, NetBeans found the executable here: `/usr/local/bin/eslint`. When I open a JS file, I got one warning and 2 infos:

```
WARNING: [se.jocke.nb.eslint.task.ESLintTaskScanner]: Callback null!!!!!
INFO: [se.jocke.nb.eslint.ESLint]: Running command /usr/local/bin/eslint
INFO: [se.jocke.nb.eslint.ESLint]: Using config /Users/myuser/.eslintrc
```

So no error, no exception. It seems to work. I didn't configured the IDE with eslint yet but I guess it will work.

**Comment by Chris2011** [comment]
So now it is needed to find the root cause. @pedrohidalgo please share a screenshot, how your input field looks like, I mean which path it was set or which path did you added manually. The button "search" finds the executable/binary automatically. I prefer to create a PR to remove that field, because it is not needed at all. I can prove that it is working w/o this field.

**Comment by Chris2011** [comment]
Node executable runs under: `/usr/local/bin/node` npm under: `/usr/local/bin/npm`. Coming from what NetBeans found.

**Comment by Chris2011** [comment]
So if no one complains, will give examples or help, I will remove this thing due to unecessary use as explained.

**Comment by Chris2011** [comment]
@joakim-eriksson please create a new verison 1.2.0 and all Mac users please test it.

**Comment by Chris2011** [comment]
Version 1.2.0 is still missing. Can you please create one @joakim-eriksson
